src/bin/rustfmt.rs

```diff
@@ -9,17 +9,37 @@
 // except according to those terms.
 
 #![cfg(not(test))]
+#![feature(path_ext)]
 
 extern crate rustfmt;
 
 use rustfmt::{WriteMode, run};
 
-use std::fs::File;
-use std::io::Read;
+use std::env;
+use std::fs::{File, PathExt};
+use std::io::{self, Read, Error, ErrorKind};
+use std::path::PathBuf;
+
+fn lookup_config_file() -> io::Result<PathBuf> {
```

Comment:
So, I was going to take this PR for now - I assumed you wanted a more flexible default.toml *and* a per-project config. But, if you also favour a hardcoded default + flexible per-project, then I think I makes sense to go straight to that, rather than take this PR in the meantime (since the end goal would mean losing some flexibility relative to this design).

I think the interesting aspect of having multiple config files is merging the per-project file with the default file (prioritising appropriately). That merging gets more complicated if you also have per-directory files (a possible compromise is per-crate files which add a little more complexity but not much). I'd be really keen for you to implement that! What do you think?